File: src/main/java/um/haberes/rest/model/DesignacionTipo.java
/**
 * 
 */
package um.haberes.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

/**
 * @author daniel
 *
 */
@Entity
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class DesignacionTipo extends Auditable implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = 7761936207749645488L;

	@Id
	private Integer designacionTipoId = 0;

	private String nombre = "";
	private BigDecimal horasSemanales = BigDecimal.ZERO;
	private BigDecimal horasTotales = BigDecimal.ZERO;
	private Integer simples = 0;

}

File: src/main/java/um/haberes/rest/kotlin/model/DesignacionTipo.kt
package um.haberes.rest.kotlin.model

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.math.BigDecimal

@Entity
@Table
data class DesignacionTipo(

    @Id
    var designacionTipoId: Int = 0,

    var nombre: String = "",
    var horasSemanales: BigDecimal = BigDecimal.ZERO,
    var horasTotales: BigDecimal = BigDecimal.ZERO,
    var simples: Int = 0

) : Auditable()
